s3-source-connector/README.md

```diff
@@ -21,7 +21,290 @@ published into the corresponding Kafka topic.
 
 The connector requires Java 11 or newer for development and production.
 
-## TODO update documentation
+#### Authorization
+
+The connector needs the following permissions to the specified bucket:
+* ``s3:GetObject``
+* ``s3:ListObjectsV2``
+
+In case of ``Access Denied`` error, see https://aws.amazon.com/premiumsupport/knowledge-center/s3-troubleshoot-403/
+
+#### Authentication
+
+To make the connector work, a user has to specify AWS credentials that allow writing to S3.
+There are two ways to specify AWS credentials in this connector:
+
+1) Long term credentials.
+
+   It requires both `aws.access.key.id` and `aws.secret.access.key` to be specified.
+2) Short term credentials.
+
+   The connector will request a temporary token from the AWS STS service and assume a role from another AWS account.
+   It requires `aws.sts.role.arn`, `aws.sts.role.session.name` to be specified.
+3) Use default provider chain or custom provider
+
+   If you prefer to use AWS default provider chain, you can leave {`aws.access.key.id` and `aws.secret.access.key`} and
+   {`aws.sts.role.arn`, `aws.sts.role.session.name`} blank. In case you prefer to build your own custom
+   provider, pass the custom provider class as a parameter to `aws.credential.provider`
+
+It is important not to use both 1 and 2 simultaneously.
+Using option 2, it is recommended to specify the S3 bucket region in `aws.s3.region` and the
+corresponding AWS STS endpoint in `aws.sts.config.endpoint`. It's better to specify both or none.
+It is also important to specify `aws.sts.role.external.id` for the security reason.
+(see some details [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_create_for-user_externalid.html)).
+
+### File name format
+
+> File name format is tightly related to [Record Grouping](#record-grouping)
+
+The connector uses the following format for input files (blobs):
+`<prefix><filename>`.
+
+`<prefix>`is the optional prefix that can be used, for example, for
+subdirectories in the bucket.
+`<filename>` is the file name. The connector has a fixed
+template for file names.
+
+    Fixed template for file : `{{topic}}-{{partition}}-{{start_offset}}`
+
+Example object name : customertopic-00001-1734445664111.txt
+
+## Data Format
+
+Connector class name, in this case: `io.aiven.kafka.connect.s3.AivenKafkaConnectS3SourceConnector`.
+
+### S3 Object Names
+
+S3 connector stores series of files in the specified bucket.
+Each object is named using pattern `[<aws.s3.prefix>]<topic>-<partition>-<start_offset>-<extension>`
```

Comment:
I dont think we have it set yet, but there should be a configuration for setting object names.

Reply:
I have rephrased it. But we are still following the pattern [<aws.s3.prefix>]<topic>-<partition>-<start_offset>-<extension> right ?